Makefile

```diff
@@ -24,37 +81,56 @@ all: install
 help:
 	@egrep "#\starget:" [Mm]akefile  | sed 's/\s-\s/\t\t\t/' | cut -d " " -f3- | sort -d
 
+# target: clean - Remove build files.
+clean:
+	$(RM) -r $(BUILD_DIR)
+
+# target: uninstall - Uninstall ALL files from the install targets.
+uninstall:
+	@for file in $(INSTALLED_FILES); do \
+			echo $(RM) $$file; \
+			$(RM) $$file; \
+	done
+
+# To change the installation root path, set the PREFIX variable in your shell's environment, like:
+# $ PREFIX=/usr/local make install
+# $ PREFIX=/tmp/test make install
 # target: install - Install all files
 install: install-scripts install-conf install-systemd
 
-
+# Install targets - add build sources to prereqa as well, so that build dir is re-created if deleted (expected behaviour).
 # target: install-scripts - Install executables.
-install-scripts:
-	install -d $(DEST_SCRIPTS)
-	install -m 0744 $(filter-out %/resticw, $(SRCS_SCRIPTS)) $(DEST_SCRIPTS)
-	install -m 0755 usr/local/sbin/resticw $(DEST_SCRIPTS)
+install-scripts: $(DEST_SCRIPTS)
+# target: install-conf - Install restic configuration files.
+install-conf: $(DEST_CONF) $(BUILD_SRCS_CONF)
+# target: install-systemd - Install systemd timer and service files.
+install-systemd: $(DEST_SYSTEMD)
 
-# Copy templates to new files with restricted permissions.
-# Why? Because the non-template files are git-ignored to prevent that someone who clones or forks this repo checks in their sensitive data like the B2 password!
-etc/restic/_global.env etc/restic/default.env etc/restic/pw.txt:
-	install -m 0600 $@.template $@
 
-# target: install-conf - Install restic configuration files.
-# will create these files locally only if they don't already exist
-# `|` means that dependencies are order-only, i.e. only created if they don't already exist.
-install-conf: | $(SRCS_CONF)
-	install -d $(DEST_CONF)
-	install -b -m 0600 $(SRCS_CONF) $(DEST_CONF)
-	$(RM) etc/restic/_global.env etc/restic/default.env etc/restic/pw.txt
+# Copies sources to build directory & replace "$INSTALL_PREFIX"
+# dir= line needs to be in the same subshell to use shared envvars. Reference: https://stackoverflow.com/a/36419671/265508
+$(BUILD_DIR)/% : %
+	${MKDIR_PARENTS} $@
+	cp $< $@
+	sed -i.bak -e "s|\$$INSTALL_PREFIX|$$PREFIX|g" $@; rm $@.bak
```

Comment:
Idea: Maybe using single quote `'` instead of `"` to clearly state that the shell will not interpret `INSTALL_PREFIX` as a variable?

Reply:
There's a tricky thing here: in a `Makefile`, make itself will try to use`$` to expand macros. To avoid this to happen, so that we can use a `$` for a shell variable, it must be escaped with another dollar sign before

After Make has unescaped the line above, it becomes this shell command line:
```bash
sed -i.bak -e "s|\$INSTALL_PREFIX|$PREFIX|g" $@; rm $@.bak
```

Because we need `$PREFIX` to expand in the double-quoted string, but not `$INSTALL_PREFIX`, the dollar is escaped in the latter. 

I see now that this is unnecessary, the value of `PREFIX` can be expanded directly from the macro instead of later in the shell expression. Then we can also use single quotes like you suggested!


I updated it to the simpler
```bash
sed -i.bak -e 's|$$INSTALL_PREFIX|$(PREFIX)|g' $@; rm $@.bak
```
now

Thanks!

Reply:
:clap: Nice! Thx to let me know, that was quite tricky indeed :) Tomorrow I'll try to test the systemd installation.